core/src/com/fourinachamber/fortyfive/screen/general/CustomActors.kt

```diff
@@ -533,6 +532,282 @@ open class CustomFlexBox(
+            scrollbarBackground.y = y
+            scrollbarBackground.width = scrollbarWidth
+            scrollbarBackground.height = height
+        } else {
+            if (scrollbarSide != null && scrollbarSide == "top") {
+                scrollbarBackground.y = y + height - scrollbarWidth
+            } else {
+                scrollbarBackground.y = y
+            }
+            scrollbarBackground.x = x
+            scrollbarBackground.width = width
+            scrollbarBackground.height = scrollbarWidth
+        }
+    }
+
+    private fun layoutScrollbarHandle() {
+        if (isScrollDirectionVertical) {
+            val max = lastMax + cutBottom
+            val curSize = height * height / (max + height)
+            val curPos = offset / max * (height - curSize)
+            if (scrollbarSide != null && scrollbarSide == "left") {
+                scrollbarHandle.x = x
+            } else {
+                scrollbarHandle.x = x + width - scrollbarWidth
+            }
+            scrollbarHandle.width = scrollbarWidth
+            scrollbarHandle.height = curSize
+            scrollbarHandle.y = y + height - curPos - curSize
+        } else {
+            val offset = offset + cutLeft
+            val max = lastMax
+            val curSize = width * width / (max + width)
+
+            val curPos = offset / max * (width - curSize)
+
+            if (scrollbarSide != null && scrollbarSide == "top") {
+                scrollbarHandle.y = y + height - scrollbarWidth
+            } else {
+                scrollbarHandle.y = y
+            }
+            scrollbarHandle.height = scrollbarWidth
+            scrollbarHandle.width = curSize
+            scrollbarHandle.x = x - curPos
+        }
+    }
+
+    override fun draw(batch: Batch?, parentAlpha: Float) {
+        batch ?: return
+        batch.flush()
+        val viewport = screen.stage.viewport
+        val xPixel = (Gdx.graphics.width - viewport.leftGutterWidth - viewport.rightGutterWidth) / viewport.worldWidth
+        val yPixel =
+            (Gdx.graphics.height - viewport.topGutterHeight - viewport.bottomGutterHeight) / viewport.worldHeight
+        if (isBackgroundStretched) {
+            if (drawBackgroundStretched(batch, xPixel, viewport, yPixel, parentAlpha)) return
+        } else {
+            super.draw(batch, parentAlpha)
+        }
+        val scissor = Rectangle(
+            xPixel * (x + cutLeft) + viewport.leftGutterWidth,
+            yPixel * (y + cutBottom) + viewport.topGutterHeight,
+            xPixel * (width - cutLeft - cutRight),
+            yPixel * (height - cutTop - cutBottom)
+        )
+        if (!ScissorStack.pushScissors(scissor)) return
+        batch.flush()
+        ScissorStack.popScissors()
+
+        batch.flush()
+        if (needsScrollbar) {
+            if (this::scrollbarBackground.isInitialized) scrollbarBackground.draw(batch, alpha)
+            if (this::scrollbarHandle.isInitialized) scrollbarHandle.draw(batch, alpha)
```

Comment:
Could scrollbarBackground and scrollbarHandle just be nullable instead of lateinit?